c/tskit/trees.c

```diff
@@ -4216,6 +4218,7 @@ tsk_tree_remove_branch(
     parent[c] = TSK_NULL;
     left_sib[c] = TSK_NULL;
     right_sib[c] = TSK_NULL;
+    self->num_children[p]--;
```

Comment:
We should follow the same pattern above and declare a local ``restrict`` pointer for ``num_children``. This is to help the compiler optimise the code, avoiding some pointer chasing. (Probably doesn't make much difference in practice, but no point in not doing it here.) 

Similarly below.